include/hipSYCL/runtime/cuda/cuda_module.hpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * This file is part of hipSYCL, a SYCL implementation based on CUDA/HIP
+ *
+ * Copyright (c) 2021 Aksel Alpay
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef HIPSYCL_CUDA_MODULE_HPP
+#define HIPSYCL_CUDA_MODULE_HPP
+
+#include <vector>
+#include <string>
+
+#include "hipSYCL/runtime/error.hpp"
+#include "hipSYCL/runtime/device_id.hpp"
+#include "hipSYCL/glue/generic/module.hpp"
+
+struct CUmod_st;
+
+namespace hipsycl {
+namespace rt {
+
+using cuda_module_id_t = unsigned long long;
```

Comment:
any reason for not using `std::uint64_t` in these occasions?

Reply:
The idea was to have as little dependencies as possible in the generated header file for robustness reasons (apart from the `std::array` and `std::string` which are needed for obvious reasons). The definition here then just aligns with what `syclcc` generates.

Reply:
okay.. that's a valid reason :) and `unsigned long long` is a good type, then (as it's guaranteed to be 64bit :))